src/main/java/core/basesyntax/service/impl/StorageAdditionServiceImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.StorageService;
+import core.basesyntax.service.TransactionLog;
+
+public class StorageAdditionServiceImpl implements StorageService {
+
+    @Override
+    public void actionToStorage(TransactionLog transactionLog, Storage storage) {
+        for (Fruit fruit : storage.getShopStorage().keySet()) {
+            if (fruit.getName().equals(transactionLog.getFruitName())) {
+                int oldValue = storage.getShopStorage().get(fruit);
+                storage.getShopStorage()
+                        .replace(fruit, oldValue + transactionLog.getFruitQuantity());
+            }
+        }
+    }
```

Comment:
same here
We don't need to iterate on map

Reply:
> same here We don't need to iterate on map

I have no idea how to do it without iteration, because input parameter is String (fruit name), but map keys are Fruit as object.

Reply:
Sorry, you are right. In this case, no cycles - no way